### ISSUE TITLE
Tests fails on 3.1.0-m3

### ISSUE THREAD

**Issue by gkorland** [xetorthio/jedis#1996]
The following tests fail on 3.1.0-m3

```
Failed tests: 
  SSLJedisClusterTest>JedisClusterTest.testLocalhostNodeNotAddedWhen127Present:615 expected:<3> but was:<0>
  ListCommandsTest.lrange:134 expected:<[]> but was:<null>
  MigrateTest.migrateConflict:104
Tests in error: 
  MigrateTest.migrate:71 » JedisData ERR Target instance replied with error: NOA...
  MigrateTest.migrateCopyReplace:127 » JedisData ERR Target instance replied wit...
  MigrateTest.migrateMulti:84 » JedisData ERR Target instance replied with error...
  SetCommandsTest.spopWithCount:155 » NullPointer list
```

**Comment by gkorland** [comment]
@sazzad16 @marcosnils  can you give  a look at 
```
  SetCommandsTest.spopWithCount:155 » NullPointer list
```
It seems very strange I don't see how this test has ever pass.

It looks like the following line should always throw NPE
```
155:     members = jedis.spop("foo", 2);
```

**Comment by sazzad16** [comment]
@gkorland Well, it's about #1961 

FYI, [my comment](https://github.com/xetorthio/jedis/pull/1961#issuecomment-487250856)
> Redis has been changing behaviour of some commands in unstable branch, compared to the latest stable branch, 5.0.

What if we use branch 5.0 for the purpose/sake of *release*? WDYT?

**Comment by gkorland** [comment]
Why do you think the behavior has change in Redis? I just tried it with 5.0.4 and 5.0.0

**Comment by sazzad16** [comment]
Because, in #1961,  tests were failing with Redis `unstable` branch but passing with `5.0` branch.

**Comment by sazzad16** [comment]
@gkorland To my understanding,

in branch `5.0`:

    CONFIG SET REQUIREPASS ''

command would **unset the password**.

But in branch `unstable`, same command results in **setting `''` as password**.

Related code in MigrateTest:

https://github.com/xetorthio/jedis/blob/5a016925ac2892877276065b505705cf2494905a/src/test/java/redis/clients/jedis/tests/commands/MigrateTest.java#L49

**Comment by sazzad16** [comment]
@gkorland about https://github.com/xetorthio/jedis/issues/1996#issuecomment-502647415:

> It seems very strange I don't see how this test has ever passed.
> It looks like the following line should always throw NPE
> 
> ```
> 155:     members = jedis.spop("foo", 2);
> ```

Previously, SPOP with COUNT option would never return `null` but empty list. But now it's possible that Redis would return null.

PS: same thing is true for LRANGE.

**Comment by sazzad16** [comment]
@gkorland 

1. PR #2004 fixes

```
ListCommandsTest.lrange:134 expected:<[]> but was:<null>
```

2. PR #2005 fixes
```
SetCommandsTest.spopWithCount:155 » NullPointer list
```

3. PR #2006 fixes
```
MigrateTest.migrateConflict:104
MigrateTest.migrate:71 » JedisData ERR Target instance replied with error: NOA...
MigrateTest.migrateCopyReplace:127 » JedisData ERR Target instance replied wit...
MigrateTest.migrateMulti:84 » JedisData ERR Target instance replied with error...
```

Hope you'd be able to review those PRs at your earliest convenience. Thanks.

**Comment by sazzad16** [comment]
Resolved by combination of #2004 , #2005 , #2006 .
